src/core/analyzer.ts

```diff
@@ -95,15 +95,18 @@ export class Analyzer {
    */
   async analyze(urls: string[]): Promise<Analysis> {
     const previousAnalysisComplete = this._analysisComplete;
+    const uiUrls = this.brandUserInputUrls(urls);
     this._analysisComplete = (async () => {
       const previousContext = await previousAnalysisComplete;
-      return await previousContext.analyze(this.brandUserInputUrls(urls));
+      return await previousContext.analyze(uiUrls);
     })();
     const context = await this._analysisComplete;
-    const results = new Map(this.brandUserInputUrls(urls).map(
-        (url) => [url, context.getDocument(context.resolveUrl(url))] as
-                     [string, Document | Warning]));
-    return new Analysis(results);
+    const resolvedUrls = uiUrls.map((url) => context.resolveUrl(url));
+    const results = new Map(resolvedUrls.map((url): [
+      ResolvedUrl,
+      Document|Warning
+    ] => [url, context.getDocument(url)]));
```

Comment:
I had suggested the cast in a recent PR because I find this combination of arrow function, type declaration, return expression and formatting really, really hard to read.